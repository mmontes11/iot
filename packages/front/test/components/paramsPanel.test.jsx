import React from "react";
import ParamsPanel from "components/paramsPanel";
import { shallow } from "enzyme";

describe("components/paramsPanel", () => {
  it("renders with some params", () => {
    const wrapper = shallow(
      <ParamsPanel
        params={[
          {
            key: "type",
            label: "Select type",
            items: ["foo", "bar"],
            isActive: false,
            isLoading: false,
            isDisabled: false,
            onButtonClick: () => undefined,
            onItemClick: () => undefined,
          },
          {
            key: "observation",
            label: "Select observation",
            items: ["foo", "bar"],
            isActive: false,
            isLoading: false,
            isDisabled: false,
            onButtonClick: () => undefined,
            onItemClick: () => undefined,
          },
          {
            key: "groupBy",
            label: "Group by",
            items: ["foo", "bar"],
            isActive: false,
            isLoading: false,
            isDisabled: false,
            onButtonClick: () => undefined,
            onItemClick: () => undefined,
          },
        ]}
        reset={{
          isDisabled: true,
          onReset: () => undefined,
        }}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
