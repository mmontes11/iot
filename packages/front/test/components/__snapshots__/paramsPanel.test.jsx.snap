// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paramsPanel renders with some params 1`] = `
<div
  className="box"
>
  <div
    className="columns"
  >
    <div
      className="column"
      key="type"
    >
      <Dropdown
        buttonStyle=""
        iconStyle="fa-angle-down"
        isActive={false}
        isDisabled={false}
        isLoading={false}
        items={
          Array [
            "foo",
            "bar",
          ]
        }
        label="Select type"
        onButtonClick={[Function]}
        onItemClick={[Function]}
      />
    </div>
    <div
      className="column"
      key="observation"
    >
      <Dropdown
        buttonStyle=""
        iconStyle="fa-angle-down"
        isActive={false}
        isDisabled={false}
        isLoading={false}
        items={
          Array [
            "foo",
            "bar",
          ]
        }
        label="Select observation"
        onButtonClick={[Function]}
        onItemClick={[Function]}
      />
    </div>
    <div
      className="column"
      key="groupBy"
    >
      <Dropdown
        buttonStyle=""
        iconStyle="fa-angle-down"
        isActive={false}
        isDisabled={false}
        isLoading={false}
        items={
          Array [
            "foo",
            "bar",
          ]
        }
        label="Group by"
        onButtonClick={[Function]}
        onItemClick={[Function]}
      />
    </div>
    <div
      className="column is-2"
    >
      <button
        className="button is-warning is-fullwidth"
        disabled={true}
        onClick={[Function]}
      >
        <FormattedMessage
          id="Reset"
          values={Object {}}
        />
        <span
          className="icon is-small"
        >
          <i
            aria-hidden="true"
            className="fas fa-eraser"
          />
        </span>
      </button>
    </div>
  </div>
</div>
`;
